Sources/Widgets/Representations/ArrowHandleRepresentation/index.js

```diff
@@ -231,83 +210,42 @@ function vtkArrowHandleRepresentation(publicAPI, model) {
     return orientationRotation;
   }
 
-  const superGetRepresentationStates = publicAPI.getRepresentationStates;
-  publicAPI.getRepresentationStates = (input = model.inputData[0]) =>
-    superGetRepresentationStates(input).filter(
-      (state) => state.getOrigin?.() && state.isVisible?.()
-    );
-
-  publicAPI.requestDataInternal = (inData, outData) => {
-    const list = publicAPI.getRepresentationStates(inData[0]);
-    const totalCount = list.length;
-
-    const points = publicAPI
-      .allocateArray('points', 'Float32Array', 3, totalCount)
-      .getData();
-    const color = publicAPI
-      .allocateArray('color', ...publicAPI.computeColorType(list), totalCount)
-      .getData();
-    const scale = publicAPI
-      .allocateArray('scale', 'Float32Array', 1, totalCount)
-      .getData();
-    const direction = publicAPI
-      .allocateArray('direction', 'Float32Array', 9, totalCount)
-      .getData();
-
-    for (let i = 0; i < totalCount; i++) {
-      const state = list[i];
-      const isActive = state.getActive();
-      const scaleFactor = isActive ? model.activeScaleFactor : 1;
-
-      const coord = state.getOrigin();
-      if (coord) {
-        points[i * 3 + 0] = coord[0];
-        points[i * 3 + 1] = coord[1];
-        points[i * 3 + 2] = coord[2];
-
-        let scale3 = state.getScale3 ? state.getScale3() : [1, 1, 1];
-        scale3 = scale3.map((x) => (x === 0 ? 2 * model.defaultScale : 2 * x));
-
-        const rotation = getGlyphRotation(scale3);
-
-        direction.set(rotation, 9 * i);
-        scale[i] =
-          scaleFactor *
-          (state.getScale1 ? state.getScale1() : model.defaultScale);
-
-        if (publicAPI.getScaleInPixels()) {
-          scale[i] *= getPixelWorldHeightAtCoord(
-            coord,
-            model.displayScaleParams
-          );
-        }
-
-        typedArray.color[i] =
-          model.useActiveColor && isActive
-            ? model.activeColor
-            : state.getColor();
-      }
+  function applyOrientation(polyData, states) {
+    model.pipeline.mapper.setOrientationArray('orientation');
+    model.pipeline.mapper.setOrientationMode(OrientationModes.MATRIX);
+    const orientation = allocateArray(
+      polyData,
+      'orientation',
+      states.length,
+      'Float32Array',
+      9
+    ).getData();
+    const defaultScale3 = [1, 1, 1];
+    for (let i = 0; i < states.length; ++i) {
+      const scale3 = states[i].getScale3?.() ?? defaultScale3;
+      const rotation = getGlyphRotation(scale3);
+      orientation.set(rotation, 9 * i);
     }
-
-    model.internalPolyData.modified();
-    outData[0] = model.internalPolyData;
-  };
+  }
+  publicAPI.setDirection(applyOrientation);
+  publicAPI.setNoOrientation(applyOrientation);
 
   publicAPI.requestData = (inData, outData) => {
+    // FIXME: shape should NOT be mixin, but a representation property.
```

Comment:
Is this FIXME actionable in this PR, or later?

Reply:
No, it is not related to this PR. Should be fixed in another PR